### PR TITLE
405b perf script updates

### PR DESCRIPTION
**Pull request by sanandaraj5597**
Added:
1) CuDNN RMSNorm
2) Multiple DistOpt instances
3) Fusing averaging of gradients inside NCCL collectives
4) Changing default GB200 config to FSDP

### CODE REVIEW

File: scripts/performance/helpers.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def set_mcore_fsdp_configs(recipe, comm_overlap_callback_idx: int | None, tp_siz
     recipe.trainer.strategy.fsdp = "megatron"
     recipe.trainer.strategy.ddp.data_parallel_sharding_strategy = "optim_grads_params"
     # At fp32 gradient, `recipe.trainer.strategy.ddp.gradient_reduce_div_fusion` is used for fusion
-    if recipe.trainer.strategy.ddp.grad_reduce_in_fp32:
+    if recipe.trainer.plugins.grad_reduce_in_fp32:
```
Comment (guyueh1) [inline scripts/performance/helpers.py:161]: which plugins are we trying to look at here? can you handle cases when `plugins` is None, or a list, or other types? 
> Trainer [doc](https://lightning.ai/docs/pytorch/stable/common/trainer.html#init): 
plugins ([Union](https://docs.python.org/3/library/typing.html#typing.Union)[[Precision](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.plugins.precision.Precision.html#lightning.pytorch.plugins.precision.Precision), [ClusterEnvironment](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.plugins.environments.ClusterEnvironment.html#lightning.pytorch.plugins.environments.ClusterEnvironment), [CheckpointIO](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.plugins.io.CheckpointIO.html#lightning.pytorch.plugins.io.CheckpointIO), [LayerSync](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.plugins.LayerSync.html#lightning.pytorch.plugins.LayerSync), [list](https://docs.python.org/3/library/stdtypes.html#list)[[Union](https://docs.python.org/3/library/typing.html#typing.Union)[[Precision](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.plugins.precision.Precision.html#lightning.pytorch.plugins.precision.Precision), [ClusterEnvironment](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.plugins.environments.ClusterEnvironment.html#lightning.pytorch.plugins.environments.ClusterEnvironment), [CheckpointIO](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.plugins.io.CheckpointIO.html#lightning.pytorch.plugins.io.CheckpointIO), [LayerSync](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.plugins.LayerSync.html#lightning.pytorch.plugins.LayerSync)]], [None](https://docs.python.org/3/library/constants.html#None)]) – Plugins allow modification of core behavior like ddp and amp, and enable custom lightning plugins. Default: None.

File: scripts/performance/llm/pretrain_llama31_405b.py [hunk 2]
```diff
@@ -170,6 +177,10 @@ def override_recipe_configs(
     exp_config = f"{num_nodes}nodes_tp{tp_size}_pp{pp_size}_cp{cp_size}_vp{vp_size}_{mbs}mbs_{gbs}gbs"
     exp_name = f"{splitext(basename(__file__))[0]}_{args.compute_dtype}_{exp_config}"
 
+    if use_mcore_fsdp:
+        # Needed to enable CuDNN LN for FSDP overlap
+        env_vars = {"NVTE_NORM_FWD_USE_CUDNN": "1", "NVTE_NORM_BWD_USE_CUDNN": "1"}
```
Comment (guyueh1) [inline scripts/performance/llm/pretrain_llama31_405b.py:179]: else: env_vars = {}

File: scripts/performance/helpers.py (hunk 1 above)
Comment (guyueh1) [reply scripts/performance/helpers.py:161]: nvm, this is ok

**Review by guyueh1 — APPROVED** [review]
LGTM
